src/routes/profile.ts

```diff
@@ -466,37 +467,36 @@ const loginUrl = url.format({
 });
 
 // show the edit page
-router.get('/:profileUsername/edit', checkProfileOwnership, (req, res) => {
-  User.findOne(
-    { usernameLower: req.params.profileUsername.toLowerCase() },
-    (err, foundUser) => {
-      if (err) {
-        console.log(err);
-      } else if (foundUser.patreonId) {
-        PatreonId.findOne({ id: foundUser.patreonId })
-          .exec()
-          .then((patreonIdObj) => {
-            res.render('profile/edit', {
-              userData: foundUser,
-              patreonLoginUrl: loginUrl,
-              patreonId: patreonIdObj,
-            });
-          })
-          .catch((err) => {
-            res.render('profile/edit', {
-              userData: foundUser,
-              patreonLoginUrl: loginUrl,
-            });
-          });
-      } else {
-        res.render('profile/edit', {
-          userData: foundUser,
-          patreonLoginUrl: loginUrl,
-        });
-      }
-    },
-  );
-});
+router.get(
+  '/:profileUsername/edit',
+  checkProfileOwnership,
+  async (req, res) => {
+    // TODO-kev: Find a better way to do the below
+    const patreonLoginUrl = patreonAgent.getPatreonAuthorizationUrl();
+    const patreonLoginUrlParams = new URLSearchParams(
```

Comment:
As discussed let's move this to the patreon routes file and have the button attempt to reach out to initiate this stuff.